crypto/src/merkle/proofs.rs

```diff
@@ -343,3 +505,15 @@ impl<H: Hasher> BatchMerkleProof<H> {
 fn are_siblings(left: usize, right: usize) -> bool {
     left & 1 == 0 && right - 1 == left
 }
+
+/// Computes the Merkle path from the computed (partial) tree.
+pub fn get_path<H: Hasher>(index: usize, tree: &BTreeMap<usize, <H as Hasher>::Digest>, depth: usize) -> Vec<H::Digest> {
+    let mut index = index + (1 << depth);
+    let mut proof = vec![*tree.get(&index).unwrap()];
```

Comment:
nit: I'd prefer not to use `.unwrap()` (here and two lines below) in non-test code. If we know that `get` will always succeed, we could use `.expect()` instead. Or, if `get` could fail, we should return an error instead.

Reply:
Is it Ok to add another error variant in the `MerkleTreeError` enum? I think it makes sense to return an error of the kind "partial tree was malformed" as it is the most appropriate in this situation.
Also I forgot but this helper function shouldn't be `pub`.

Reply:
I'd probably just return `InvalidProof` variant. My assumption is that partial tree would end up being malformed only if the original batch proof was malformed, right?

Reply:
Yes, if it is used exactly like we do now inside `into_paths`.